### ISSUE TITLE
NullPointerException on build (IEP-811)

### ISSUE THREAD

**Issue by Pieter12345** [espressif/idf-eclipse-plugin#634]
**Describe the bug**
After completing the ESP-IDF installation using the offline Windows installer, I found that every time I run a build, it ends in the following exception:
![afbeelding](https://user-images.githubusercontent.com/9693840/202240735-ceea8c1d-1ff1-4b0a-8f12-bbfb1a17ce5d.png)
I've tried updating using the beta update site, but the error remained the same.
Possibly related to #61.
This error prevents me from flashing programs to my ESP32.

**To Reproduce**
1. Run the offline ESP-IDF installer.
2. Change the Python path in the config to use included Python 3.8 instead of the non-included 3.11 (this is a fix for a separate bug).
3. Import any project (I used https://github.com/espressif/esp-idf-template) using `Import -> Existing IDF Project`.
4. Open a project file and press `Build` (the left-top hammer icon). The build runs, information about the used DRAM, IRAM and flash size is printed (looks like a succesful build) and the error appears in a popup.

**Expected behavior**
I expect this error not to happen.

**Screenshots**
![afbeelding](https://user-images.githubusercontent.com/9693840/202242869-9afc3a09-f66a-477e-9e7d-d85a1f93d86d.png)

**Build feedback**
```
Building in: C:\Users\Pietje\Documents\My-Workspace\ESP32-Workspace\esp-idf-template\build
cmake --build . -- -v
[1/4] cmd.exe /C "cd /D C:\Users\Pietje\Documents\My-Workspace\ESP32-Workspace\esp-idf-template\build\esp-idf\esptool_py && python C:/Users/Pietje/Programs/ESP32IDE/frameworks/esp-idf-v4.4.3/components/partition_table/check_sizes.py --offset 0x8000 partition --type app C:/Users/Pietje/Documents/My-Workspace/ESP32-Workspace/esp-idf-template/build/partition_table/partition-table.bin C:/Users/Pietje/Documents/My-Workspace/ESP32-Workspace/esp-idf-template/build/app-template.bin"
app-template.bin binary size 0x299a0 bytes. Smallest app partition is 0x100000 bytes. 0xd6660 bytes (84%) free.
[2/4] cmd.exe /C "cd /D C:\Users\Pietje\Documents\My-Workspace\ESP32-Workspace\esp-idf-template\build\bootloader && C:\Users\Pietje\Programs\ESP32IDE\tools\cmake\3.23.1\bin\cmake.exe --build ."
[1/1] cmd.exe /C "cd /D C:\Users\Pietje\Documents\My-Workspace\ESP32-Workspace\esp-idf-template\build\bootloader\esp-idf\esptool_py && python C:/Users/Pietje/Programs/ESP32IDE/frameworks/esp-idf-v4.4.3/components/partition_table/check_sizes.py --offset 0x8000 bootloader 0x1000 C:/Users/Pietje/Documents/My-Workspace/ESP32-Workspace/esp-idf-template/build/bootloader/bootloader.bin"
Bootloader binary size 0x6330 bytes. 0xcd0 bytes (11%) free.
Build complete (0 errors, 0 warnings): C:\Users\Pietje\Documents\My-Workspace\ESP32-Workspace\esp-idf-template\build
C:\Users\Pietje\Programs\ESP32IDE\python_env\idf4.4_py3.8_env\Scripts\python.exe C:\Users\Pietje\Programs\ESP32IDE\frameworks\esp-idf-v4.4.3\tools\idf_size.py C:/Users/Pietje/Documents/My-Workspace/ESP32-Workspace/esp-idf-template/build/app-template.map
Total sizes:

Used static DRAM:   11248 bytes ( 169488 remain, 6.2% used)

      .data size:    8968 bytes

      .bss  size:    2280 bytes

Used static IRAM:   46270 bytes (  84802 remain, 35.3% used)

      .text size:   45243 bytes

   .vectors size:    1027 bytes

Used Flash size :  115023 bytes

      .text     :   84115 bytes

      .rodata   :   30652 bytes

Total image size:  170261 bytes (.bin may be padded larger)

C:\Users\Pietje\Programs\ESP32IDE\python_env\idf4.4_py3.8_env\Scripts\python.exe C:/Users/Pietje/Programs/ESP32IDE/frameworks/esp-idf-v4.4.3/\components\partition_table\gen_esp32part.py C:\Users\Pietje\Documents\My-Workspace\ESP32-Workspace\esp-idf-template\build\partition_table\partition-table.bin
Total time taken to build the project: 650 ms
```

**Espressif-IDE Product Information:**
[config.txt](https://github.com/espressif/idf-eclipse-plugin/files/10023993/config.txt)

**Eclipse Error log:** 
Window > Show View > Other > Search for "Error Log" (Attach as a file)
[error.log](https://github.com/espressif/idf-eclipse-plugin/files/10024041/error.log)

**Comment by sigmaaa** [comment]
Hi @Pieter12345,

Thank you for reporting this issue. Can you check if the project nature is idfNature? 
To check it: 
Right-click on a project in project explorer -> Properties -> Project Natures:
![image](https://user-images.githubusercontent.com/24419842/202301102-ec625c44-9c2b-4dd7-b30f-586eaa96705c.png)

To make sure it's selected, click it, then click Apply and close. Let us know if this helps. thank you

**Comment by Pieter12345** [comment]
I found that the project had illegal settings (Eclipse showed empty project settings in the GUI with an error about them containing invalid values). I've removed my project files and re-imported the project in the current version of the IDE and now it appears to work just fine. It should be noted that the recommended offline installer for Windows does not (always?) result in a succesful installation. I've spent so much time just to set this development environment up.

**Comment by sigmaaa** [comment]
Hi @Pieter12345 

I understand your frustration and I'm sorry you're having this problem. 

When installing Espressif-IDE from the online or offline installer, the environment must be configured based on the settings selected during installation. I was trying to reproduce your issue by installing fresh Espressif-IDE from the installer and can't confirm, that I was experiencing the same issue. 

Just after the fresh installation, I imported a freshly cloned project https://github.com/espressif/esp-idf-template and the build was successful. Also tried to build a few newly created and the build was fine as well. The only time I've run into something like this is when I imported a project that contained a CDT Builder Project nature, but it happened because the nature wasn't added correctly. Was your project modified in any way before import? This information will help us reproduce this problem. Thank you
